index.js

```diff
@@ -37,10 +38,12 @@ function startServer(config, logger) {
 
     app.use((req, res, next) => handleNotFound(req, res, next, logger));
 
-    const listener = app.listen(port, () => {
-        logger.info(`Started server on port ${listener.address().port}`);
-        opn(`http://localhost:${listener.address().port}`);
-    });
+    portfinder.getPortPromise({port})
+        .then(availablePort => app.listen(availablePort, () => {
+            logger.info(`Started server on port ${availablePort}`);
+            opn(`http://localhost:${availablePort}`);
+        }))
+        .catch(error => logger.error(`Could not find an available port: ${error}`));
```

Comment:
I think that any error that occurs within your .then() function (app.listen() fails, for example) will get caught in your catch() handler. So, you could potentially log an error that says `Could not find an available port: some express error happened unrelated to portfinder`.

Reply:
Using async/await with try/catch blocks might make this easier.